File: c++/1214 - Side of the Rhombus.cpp
#include <stdio.h>
#include <math.h>

int main(void)
{
    int t;
    double a, s, side;
    scanf("%d", &t);
    while (t--)
    {
        scanf("%lf %lf", &a, &s);
        printf("%.2f\n", sqrt(pow(s, 2) - (4 * a)) / 2);
    }
    return 0;
}

File: Java/1214 - Side of the Rhombus.java
import java.util.*;

public class coj {

  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    Double t = sc.nextDouble();

    for (int i = 0; i < t; i++) {
      Double a = sc.nextDouble();
      Double s = sc.nextDouble();

      System.out.printf("%.2f\n", Math.sqrt(Math.pow(s, 2) - (4 * a)) / 2);
    }
  }
}
